src/engine/strat_engine/mdv.rs

```diff
@@ -37,6 +37,51 @@ const FILESYSTEM_DIR: &'static str = "filesystems";
 pub struct MetadataVol {
     dev: LinearDev,
     mount_pt: PathBuf,
+    is_mounted: bool,
+}
+
+/// A helper struct that borrows the MetadataVol and ensures that the MDV is
+/// mounted as long as it is borrowed, and then unmounted.
+#[derive(Debug)]
+struct MountedMDV<'a> {
+    mdv: &'a mut MetadataVol,
+}
+
+impl<'a> MountedMDV<'a> {
+    /// Borrow the MDV and ensure it's mounted.
+    fn mount(mdv: &mut MetadataVol) -> EngineResult<MountedMDV> {
+        if !mdv.is_mounted {
+            match mount(Some(&mdv.dev.devnode()),
+                        &mdv.mount_pt,
+                        Some("xfs"),
+                        MsFlags::empty(),
+                        None as Option<&str>) {
+                Err(nix::Error::Sys(nix::Errno::EBUSY)) => {
+                    // We thought we weren't mounted but we actually were
+                    Ok(())
+                }
+                Err(err) => Err(err),
+                Ok(_) => Ok(()),
+            }?;
+
+            mdv.is_mounted = true;
+        }
+
+        Ok(MountedMDV { mdv })
+    }
+
+    fn mount_pt(&self) -> &Path {
+        &self.mdv.mount_pt
+    }
+}
+
+impl<'a> Drop for MountedMDV<'a> {
+    fn drop(&mut self) {
+        match umount(&self.mdv.mount_pt) {
+            Err(err) => warn!("Could not unmount MDV: {}", err),
```

Comment:
If unmount fails, we can panic (which I don't think is appropriate), or print a message, but we can't return an error, and we shouldn't just silently let it go, because if unmount is failing then we need to know to find out why. We have #309 to improve this en masse, and have guidelines in the style guide, but until then what should we do here?